Add template support to KafkaUser resource to manage secret metadata (#3263)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.template.KafkaUserTemplate.adoc

> `KafkaUserTemplate` can be used specify additional labels and annotation for the secret with user password or certificate created by the User Operator.

Reviewer: ```suggestion
`KafkaUserTemplate` is used specify additional labels and annotations for the secret created by the User Operator.
```
